Upload weekly autoban report to pastebin

ChrisLovering opened the pull request:
This fixes #2391 and solves BOT-3CR which was caused by the weekly report being longer than 2000 characters.

Instead of posting the entire report to Discord, it is now upload to the pastebin, and uploaded as a text file to the channel, so people can use what they prefer.

vivekashok1221 reviewed (dismissed):
Code looks good to me, will test after a few hours.
There's a typo in the docstring of `send_weekly_autoban_report` method, btw.
>    If chanel is not specified, it is sent to #mod-meta.
line 316.

mbaruh commented:
@ChrisLovering can you resolve the merge conflict? it'll be moving the changes to the equivalent function in the new filtering.py
https://github.com/python-discord/bot/blob/ea0b227801f19d14338e7e582d745ea250773370/bot/exts/filtering/filtering.py#L1396

ChrisLovering commented:
> @ChrisLovering can you resolve the merge conflict? it'll be moving the changes to the equivalent function in the new filtering.py
> 
> https://github.com/python-discord/bot/blob/ea0b227801f19d14338e7e582d745ea250773370/bot/exts/filtering/filtering.py#L1396

Rebased, I've also simplified it quite a lot, it was far more complicated than it needed to be.

wookie184 commented on bot/exts/filtering/filtering.py:
```diff
@@ -1432,7 +1434,17 @@ async def send_weekly_auto_infraction_report(self, channel: discord.TextChannel
 
-        await channel.send("\n\n".join(lines))
+        report = "\n\n".join(lines)
+        if len(report) > MAX_PASTE_LENGTH:
```
`MAX_PASTE_LENGTH` is the maximum length that can be sent to the pastebin which isn't what we want here. Not sure if we have a constant anywhere for message length.

ChrisLovering replied:
🤦 I'll sort it

mbaruh commented on bot/exts/filtering/filtering.py:
```diff
@@ -1432,9 +1434,20 @@ async def send_weekly_auto_infraction_report(self, channel: discord.TextChannel
-        await channel.send("\n\n".join(lines))
-        log.info("Successfully sent auto-infraction report.")
+        report = "\n\n".join(lines)
+        try:
+            await channel.send(report)
+        except discord.HTTPException as e:
+            if e.code == 50035:  # Content too long
+                report = discord.utils.remove_markdown(report)
+                paste_url = await send_to_paste_service(report, extension="txt")
+                file_buffer = io.StringIO(report)
+                await channel.send(
+                    f"**{lines[0]}**\n\n{paste_url}",
+                    file=discord.File(file_buffer, "last_weeks_autoban_filters.txt"),
+                )
```
Doesn't this suppress any other reasons for HTTP exceptions? you'll end up not having a report and then logging that it was sent successfully.

ChrisLovering replied:
True, I've made is so that these still get raised.

mbaruh commented on bot/exts/filtering/filtering.py:
```diff
@@ -1432,9 +1434,20 @@ async def send_weekly_auto_infraction_report(self, channel: discord.TextChannel
+            await channel.send(report)
+        except discord.HTTPException as e:
+            if e.code == 50035:  # Content too long
+                report = discord.utils.remove_markdown(report)
```
What is the concern behind removing markdown before uploading it?

mbaruh commented on bot/exts/filtering/filtering.py:
```diff
@@ -1432,9 +1434,20 @@ async def send_weekly_auto_infraction_report(self, channel: discord.TextChannel
+        except discord.HTTPException as e:
+            if e.code == 50035:  # Content too long
+                report = discord.utils.remove_markdown(report)
+                paste_url = await send_to_paste_service(report, extension="txt")
```
This can raise exceptions that should be handled.
